### PR TITLE
Created Sanity tests run for Consumer Pages

### PR DESCRIPTION
**Pull request by AndreiH**
Adhoc here: http://selenium.qa.mtv2.mozilla.com:8080/view/Marketplace/job/marketplace.STAGE.saucelabs.sanity_tests.andreih/

### CODE REVIEW

**Comment by AndreiH** [comment]
We have adhocs for all 3 envs for some time now:

http://selenium.qa.mtv2.mozilla.com:8080/view/Marketplace/job/marketplace.DEV.saucelabs.sanity_tests.andreih/

http://selenium.qa.mtv2.mozilla.com:8080/view/Marketplace/job/marketplace.PROD.saucelabs.sanity_tests.andreih/

http://selenium.qa.mtv2.mozilla.com:8080/view/Marketplace/job/marketplace.STAGE.saucelabs.sanity_tests.andreih/

File: pages/desktop/consumer_pages/category.py [hunk 1]
```diff
@@ -32,10 +37,50 @@ def category_title(self):
         return self.selenium.find_element(*self._category_section_title_locator).text
 
     @property
-    def apps_count(self):
-        return self.selenium.find_elements(*self._category_items_locator)
+    def apps_number(self):
```
Comment (bobsilverberg) [inline pages/desktop/consumer_pages/category.py]: I agree with this. @AndreiH Is there a reason you didn't make this change?

**Comment by bobsilverberg** [comment]
@AndreiH The code looks fine to me, but I agree with @viorelaioia's comment, so I think you should change that.

The tests all pass for me on stage and prod, but lots of them fail on dev. This is also happening on the adhoc job [1]. That may be because there have been changes made on dev that are not reflected in this PR.

As this is so old and seems to be working on stage and prod, I'd give this an r+ to be merged, after addressing @viorelaioia's comment. We could then open separate issues to address the failures on dev, as it would be nice to get this one in.

[1] http://selenium.qa.mtv2.mozilla.com:8080/view/Marketplace/job/marketplace.DEV.saucelabs.sanity_tests.andreih/169/console

File: pages/desktop/consumer_pages/category.py (hunk 1 above)
Comment (AndreiH) [inline pages/desktop/consumer_pages/category.py]: this slipped through :) Thanks!

**Comment by AndreiH** [comment]
Adhocs seem to be passing now @bobsilverberg 
Thanks again!

**Comment by bobsilverberg** [comment]
This is all good, @AndreiH, and ready to merge except for one thing. Have you watched `test_opening_every_category_page_from_categories_menu` run? I noticed something strange when I ran it locally, and I am seeing it to on the Saucelabs run [1]. Although the test runs and passes, it doesn't seem to open *any* of the category pages, let alone all of them! So it looks like there is something seriously wrong with the test.

I'm going to merge this anyway, as it's been outstanding for a long time, but I will also open an issue to address that test.

[1] https://saucelabs.com/jobs/bbb8c652a7384fe4b7574d684cdbd58c

**Comment by bobsilverberg** [comment]
@AndreiH Actually, I'm not going to merge it as there are tons of commits and some of them are merge commits, so untangling that is going to be difficult for someone who didn't work on the code, so I'm going to ask you to squash the commits down to 1, and then this can be merged.

**Comment by AndreiH** [comment]
@bobsilverberg thanks for spotting that. <code>test_opening_every_category_page_from_categories_menu</code> was failing because <code>home_page.category_count</code> was <code>init:0</code> because the categories locator was changed. This is worrying and we need to make sure that the test will fail if locator will change again, let's do that in the issue you opened. For now I squashed the commits and updated the <code>test_opening_every_category_page_from_categories_menu</code> to run properly. Thanks!

**Comment by bobsilverberg** [comment]
Thanks for squashing @AndreiH. I will merge this and we can track the other issue in issue #610.

**Pull request merged**